ad-tag/source/ts/console/debug.tsx

```diff
@@ -35,17 +35,63 @@ if (moliConfig) {
   // attach shadow DOM to container
   const shadowRoot = container.attachShadow({ mode: 'open' });
 
-  // create react element
-  const globalConfigComponent = (
-    <GlobalConfig
-      config={moliConfig}
-      runtimeConfig={window.moli.getRuntimeConfig()}
-      modules={window.moli.getConfig()?.modules || {}}
-      labelConfigService={labelConfigService}
-      windowResizeService={new WindowResizeService()}
-      themingService={themingService}
-    />
-  );
+  // function component to manage showOverlays state
+  const DebugConsole = () => {
+    const [showOverlays, setShowOverlays] = React.useState(false);
+    const slotRoots = React.useRef(new Map<string, ReturnType<typeof createRoot>>());
+
+    React.useEffect(() => {
+      setTimeout(() => {
+        const globalConfigComponentRoot = shadowRoot.getElementById('moli-console-global-config');
+        themingService.setRootElement(globalConfigComponentRoot!);
+        themingService.applyTheme();
+        themingService.enableSystemThemeListener();
+      }, 200);
+    }, []);
+
+    React.useEffect(() => {
+      if (showOverlays) {
+        moliConfig.slots.forEach(slot => {
+          const slotDomElement = document.getElementById(slot.domId);
+          if (slotDomElement && labelConfigService.filterSlot(slot)) {
+            let slotConfigRoot = slotRoots.current.get(slot.domId);
```

Comment:
The AdSlotConfig components can unfortunately not simply be rendered as part of the React tree since this would render them inside the shadow DOM. However, the overlays must render in the actual slot DOM elements.